src/config/machines/example-grant-machine.js

```diff
@@ -99,30 +23,18 @@ export const exampleGrantMachine = createMachine({
         }
       },
       meta: {
-        // TODO: think about removing duplication with on.NEXT.target
         currentPageId: 'start',
         nextPageId: 'country'
       }
     },
 
     country: {
-      on: {
-        BACK: {
-          target: 'start',
-          actions: ['updateCurrentPageId']
-        },
-
-        NEXT: {
-          target: 'second-question', // NOSONAR:S1192 - need to improve this later
-          actions: ['trackPageCompletion', 'updateCurrentPageId', 'updateAnswers']
-        }
-      },
-
+      ...createTransitions('start', 'second-question'),
       meta: {
+        ...QUESTION_META,
         currentPageId: 'country',
```

Comment:
Like we have `createTransitions`, `createMeta` could be useful and using a lens we could potentially have something like `const [transitions, meta] = createAnswerState('country')('start', 'second-question')`.

Reply:
Agreed. This was on my deeper layer of refactor but as I said I got in a mess and decided to start afresh. Definitely one for the future.